### PR TITLE
Fixed Bug in Animations.js and implemented some features in Grid Visualizer

### PR DESCRIPTION
**Pull request by AliCodingGuru**
In this commit i fixed the bug in Animations.js line#2 by deleting the line cause due to this Js code was returning nothing and as a result no animations were played both in local environment as well as in deployed site.
I also added a feature in Grid Section by which user can copy the css code for the grid on his own site.
I added custom number of elements in grid visualizer by which user can Visualize upto 15 elements in Grid.

### CODE REVIEW

**Comment by fahdi** [comment]
@AliCodingGuru Please follow the issue template guideline and then link that with your PR. Also, need screenshots in the PR.

**Comment by AliCodingGuru** [comment]
@fahdi 
---
name: Bug report in Animations.js
about: 'Animations Visualizer' 
title: 'Bug in Animations Visualizer'
labels: ''
assignees: ''

---

**Bug Description**
This bug is caused by line 3,4,5 in animations .js which are
` if (window.location.pathname !== '/animations.html') {
    return;
  }`
Due to these lines the Js files terminates at the start by returning nothing. This is because the url of the css visual learner is according to template fahdi.github.io/css-visual-learner/{file-name} and it doesn't matches the url mentioned in JS File.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to 'https://fahdi.github.io/css-visual-learner/animations.html'
2. Click on 'apply animation' button or any button next to it. None of them will work.

**Expected behavior**
The Javascript file should run the animations on the page.

**Screenshots**
Screenshot_2023-11-29_00_15_34
Screenshot_2023-11-29_00_17_18


**Desktop (please complete the following information):**
 - OS: Kali Linux
 - Browser: Firefox
 - Version 91.8.0

**Additional context**
I have solved this problem in Pull Request by deleting line so you can check that out.

**Comment by fahdi** [comment]
@AliCodingGuru Please update the PRF description with this info you put as comment.

**Comment by fahdi** [comment]
Also, please create an issue relevant to this. I don't see it here: https://github.com/fahdi/css-visual-learner/issues

File: scripts/animations.js [hunk 1]
```diff
@@ -1,15 +1,11 @@
 // document onload event
 document.addEventListener('DOMContentLoaded', function(){
-  if (window.location.pathname !== '/animations.html') {
-    return;
```
Comment (fahdi) [inline scripts/animations.js:4]: Any particular reason for removing this?

File: scripts/grid.js [hunk 2]
```diff
@@ -4,20 +4,36 @@ document.addEventListener('DOMContentLoaded', function(){
   const gridColumnGapInput = document.getElementById('gridColumnGap');
   const gridRowGapInput = document.getElementById('gridRowGap');
   const gridContainer = document.querySelector('.grid-container');
-
+  const codeDisplayer = document.querySelector('#csscode p')
+  const CopyStyleButton = document.querySelector('#csscode button')
+  const NoOfContainers = document.querySelector('.controls input')
   function applyStyles(){
-    gridContainer.style.cssText = `
+    var css = gridContainer.style.cssText = `
```
Comment (fahdi) [inline scripts/grid.js:11]: use let instead.

File: scripts/grid.js [hunk 3]
```diff
@@ -4,20 +4,36 @@ document.addEventListener('DOMContentLoaded', function(){
...
         grid-template-columns: ${gridTemplateColumnsInput.value};
         grid-template-rows: ${gridTemplateRowsInput.value};
         column-gap: ${gridColumnGapInput.value};
         row-gap: ${gridRowGapInput.value};
       `;
+    codeDisplayer.textContent = css
+  }
+  function copycss(){
```
Comment (fahdi) [inline scripts/grid.js:19]: Add space above this? Are you linting your JS?

File: scripts/grid.js [hunk 4]
```diff
@@ -4,20 +4,36 @@ document.addEventListener('DOMContentLoaded', function(){
...
+    navigator.clipboard.writeText(gridContainer.style.cssText)
   }
 
   // Event listeners for input changes
   gridTemplateColumnsInput.addEventListener('input', applyStyles);
   gridTemplateRowsInput.addEventListener('input', applyStyles);
   gridColumnGapInput.addEventListener('input', applyStyles);
   gridRowGapInput.addEventListener('input', applyStyles);
+  NoOfContainers.addEventListener('input',() => {
```
Comment (fahdi) [inline scripts/grid.js:28]: Is that a good way to name a variable? Please change it to camelCase.

Comment (fahdi) [reply scripts/grid.js:28]: Also, don't use the arrow function, we use full declaration as of now. Let's keep the same standard.

File: styles/grid.css [hunk 5]
```diff
@@ -38,3 +44,21 @@
 .controls select:active {
     border: 1px solid blue;
 }
+#liveCssDisplayer{
```
Comment (fahdi) [inline styles/grid.css:47]: Don't use camel case for class names.

File: scripts/grid.js [hunk 6]
```diff
@@ -4,20 +4,36 @@ document.addEventListener('DOMContentLoaded', function(){
   const gridColumnGapInput = document.getElementById('gridColumnGap');
   const gridRowGapInput = document.getElementById('gridRowGap');
   const gridContainer = document.querySelector('.grid-container');
-
+  const codeDisplayer = document.querySelector('#csscode p')
+  const CopyStyleButton = document.querySelector('#csscode button')
```
Comment (fahdi) [inline scripts/grid.js:8]: camelCase please!

**Comment by fahdi** [comment]
@AliCodingGuru Any news on this?

**Pull request closed without merging**